python/cudf/dataframe/dataframe.py

```diff
@@ -1921,7 +1921,11 @@ def from_pandas(cls, dataframe, nan_as_null=True):
         df = cls()
         # Set columns
         for colk in dataframe.columns:
-            df[colk] = Series(dataframe[colk].values, nan_as_null=nan_as_null)
+            vals = dataframe[colk].values
+            # CUDF assumes values are always contiguous
+            if not vals.flags['C_CONTIGUOUS']:
+                vals = np.ascontiguousarray(vals)
```

Comment:
I think this logic should be moved into the `columnops.as_column` function so that it's used in functions other than `from_pandas` as well.